### PR TITLE
boards: add support for stm32f401_mini

### PR DESCRIPTION
**Pull request by bpbradley**
Adds support for [the STM32 Mini F401 board](https://stm32-base.org/boards/STM32F401CCU6-STM32-Mini-F401).

It is the lowest cost / most barebones board that I could find which is domestically available (can buy it on Amazon for ~5 dollars in the US).

### CODE REVIEW

**Review by FRASTM — DISMISSED** [review]

File: boards/arm/stm32f401_mini/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,155 @@
+.. _stm32f401_mini:
+
+STM32 Mini F401
+###############
+
+Overview
+********
+
+The STM32 Mini S401 is an extremely low cost and bare-bones
```
Comment (erwango) [inline boards/arm/stm32f401_mini/doc/index.rst:9]: s/S401/F401

**Review by erwango — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

File: boards/arm/stm32f401_mini/stm32f401_mini_defconfig [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_STM32F4X=y
+CONFIG_SOC_STM32F401XE=y
```
Comment (ABOSTM) [inline boards/arm/stm32f401_mini/stm32f401_mini_defconfig:4]: CONFIG_SOC_STM32F401XC=y

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f401_mini/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+Clock Sources
+-------------
+
+The board has two external oscillators. The frequency of the slow clock (LSE) is
+32.768 kHz. The frequency of the main clock (HSE) is 25 MHz.
+
+The default configuration sources the system clock from the PLL, which is
+derived from HSE, and is set at 84MHz, which is the maximum possible frequency
+to achieve a stable USB clock (42MHz).
```
[the remark this answers is not in the document]
Comment (bpbradley) [reply boards/arm/stm32f401_mini/doc/index.rst]: Fixed. This was an artifact from the docs of a board I made in the past. I simply removed it as it isn't that useful of a note anyway. Thanks for catching.

File: boards/arm/stm32f401_mini/stm32f401_mini_defconfig (hunk 2 above)
Comment (bpbradley) [reply boards/arm/stm32f401_mini/stm32f401_mini_defconfig]: Fixed

**Review by ABOSTM — CHANGES_REQUESTED** [review]

**Review by FRASTM — DISMISSED** [review]

File: boards/arm/stm32f401_mini/stm32f401_mini.dts [hunk 4]
```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2022 Brian Bradley (brian.bradley.p@gmail.com)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f4/stm32f401Xe.dtsi>
```
[the remark this answers is not in the document]
Comment (bpbradley) [reply boards/arm/stm32f401_mini/stm32f401_mini.dts:8]: Of course... I probably should have built a sample after making the last change. I will have it fixed in a few minutes.

Thanks!

**Comment by bpbradley** [comment]
It seems tests are failing because of an issue with the nucleo_l4r5zi? I don't think this is related to my PR.

**Review by erwango — DISMISSED** [review]

**Review by ABOSTM — DISMISSED** [review]

**Comment by benediktibk** [comment]
@bpbradley #43750 is fixed, ready for rebase

**Comment by bpbradley** [comment]
Looks like I am waiting on the hotfix in #43805 before this will pass

**Review by MaureenHelm — DISMISSED** [review]

**Review by ABOSTM — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Comment by bpbradley** [comment]
> 

@erwango did I mess something up? I am still having the same issue. I rebased to main and force pushed. My commit is on top of ae71444426a07a918ff82132e046e90d05093725 which I believe was supposed to resolve this issue.

**Comment by erwango** [comment]
It seems we can't rely on CI to detect issues on unrelated PRs but not much on the one we need.
Last step is https://github.com/zephyrproject-rtos/zephyr/pull/43857 (hopefully)

**Comment by bpbradley** [comment]
> 

Sounds good, I will wait for that to be merged before requesting re-reviews. Thanks!

**Review by erwango — APPROVED** [review]

**Comment by bpbradley** [comment]
Thanks @erwango  -- Looks like the fix for the CI issue was merged and tests are now passing. Re-requesting reviews now.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**